### ISSUE TITLE
Compatibility with x86-32bit

### ISSUE THREAD

**Issue by ikervagyok** [badboy/minilzo-rs#2]
I packaged btrfs-dedupe for nixos and the builds for x86 are failing, while builds for x86_64 are working. the error suggests, that minilzo is not compiling.

My request would be:
- Would it be possible to add 32bit support?
- If it is already implemented what do i need to do, to make it compile/work?

Thanks, ikervagyok

---
Here is the error, which i get when compiling btrfs-dedupe:
```
building path(s) ‘/nix/store/n160qsibkvjlkjgiqmzsbxxxknqm62n0-btrfs-dedupe-1.1.0’
unpacking sources
unpacking source archive /nix/store/i97ykw2p7bzl9lm0dhllmhc29db3d5a7-archive.tar.bz2?ref=72c6a301d20f935827b994db210bf0a1e121273a
source root is btrfs-dedupe-72c6a301d20f935827b994db210bf0a1e121273a-72c6a301d20f935827b994db210bf0a1e121273a
Using cargo deps from /nix/store/z3218s53bxxhv9vrlhvmbmq18l5wkxz6-btrfs-dedupe-1.1.0-fetch
Using indexHash '-ba82b75dd6681d6f'
Using rust registry from /nix/store/mnsaxh5zksckcar3pxz7b6yac980r08c-rustRegistry-2017-02-19-d822c58
warning: custom registry support via the `registry.index` configuration is being removed, this functionality will not work in the future
warning: custom registry support via the `registry.index` configuration is being removed, this functionality will not work in the future
setting SOURCE_DATE_EPOCH to timestamp 1488351352 of file btrfs-dedupe-72c6a301d20f935827b994db210bf0a1e121273a-72c6a301d20f935827b994db210bf0a1e121273a/Cargo.lock
warning: file btrfs-dedupe-72c6a301d20f935827b994db210bf0a1e121273a-72c6a301d20f935827b994db210bf0a1e121273a/Cargo.lock may be generated; SOURCE_DATE_EPOCH may be non-deterministic
patching sources
configuring
building
Running cargo build --release
warning: custom registry support via the `registry.index` configuration is being removed, this functionality will not work in the future
   Compiling vec_map v0.6.0 (registry file:///dev/null)
   Compiling log v0.3.6 (registry file:///dev/null)
   Compiling lazy_static v0.2.2 (registry file:///dev/null)
   Compiling rustc-serialize v0.3.22 (registry file:///dev/null)
   Compiling unicode-xid v0.0.3 (registry file:///dev/null)
   Compiling quote v0.3.10 (registry file:///dev/null)
   Compiling unicode-segmentation v1.0.1 (registry file:///dev/null)
   Compiling byte-tools v0.1.2 (registry file:///dev/null)
   Compiling bitflags v0.7.0 (registry file:///dev/null)
   Compiling crc v1.4.0 (registry file:///dev/null)
   Compiling ansi_term v0.9.0 (registry file:///dev/null)
   Compiling unicode-width v0.1.4 (registry file:///dev/null)
   Compiling dtoa v0.2.2 (registry file:///dev/null)
   Compiling itoa v0.1.1 (registry file:///dev/null)
   Compiling libc v0.2.18 (registry file:///dev/null)
   Compiling typenum v1.5.1 (registry file:///dev/null)
   Compiling uuid v0.3.1 (registry file:///dev/null)
   Compiling strsim v0.6.0 (registry file:///dev/null)
   Compiling fake-simd v0.1.2 (registry file:///dev/null)
   Compiling term_size v0.2.1 (registry file:///dev/null)
   Compiling minilzo-sys v0.1.0 (registry file:///dev/null)
   Compiling ioctl v0.3.3 (registry file:///dev/null)
   Compiling minilzo v0.1.0 (registry file:///dev/null)
   Compiling minilzo v0.1.0 (registry file:///dev/null)
error[E0308]: mismatched types
   --> /tmp/nix-build-btrfs-dedupe-1.1.0.drv-0/deps/registry/src/-ba82b75dd6681d6f/minilzo-0.1.0/src/lib.rs:121:13
    |
121 |             inlen as u64,
    |             ^^^^^^^^^^^^ expected u32, found u64

error[E0308]: mismatched types
   --> /tmp/nix-build-btrfs-dedupe-1.1.0.drv-0/deps/registry/src/-ba82b75dd6681d6f/minilzo-0.1.0/src/lib.rs:161:13
    |
161 |             inlen as u64,
    |             ^^^^^^^^^^^^ expected u32, found u64

error: aborting due to 2 previous errors

Build failed, waiting for other jobs to finish...
error: Could not compile `minilzo`.

To learn more, run the command again with --verbose.
builder for ‘/nix/store/09977dmw443ww1xhracjlipj794s11hi-btrfs-dedupe-1.1.0.drv’ failed with exit code 101
```

**Comment by badboy** [comment]
Thanks for the report. I'll take a look to get it working on 32-bit

**Comment by badboy** [comment]
Oh wait, I already fixed that and released a new version. Just upgrade the dependency.

**Comment by ikervagyok** [comment]
Thanks, so I should ask the btrfs-dedupe developer to update his crate.toml? Sorry, but I am not very familiar with rust packaging.

**Comment by badboy** [comment]
All it needs would be a rebuild of the included [`Cargo.lock`](https://gitlab.wellbehavedsoftware.com/well-behaved-software/btrfs-dedupe/blob/master/Cargo.lock), which currently is fixed to `v0.1.0` of minilzo.
Approaching the upstream dev of btrs-dedupe would be the best idea
